androguard/core/bytecodes/dvm.py

```diff
@@ -7506,35 +7469,9 @@ def __init__(self, cm, off, buff):
 
             buff.set_idx(idx + mi.get_length())
 
-        # TYPE_STRING_DATA_ITEM will be at the beginning of ordered
-        # We want to parse this first, as other map items depend on it.
+        load_order = TypeMapItem.determine_load_order()
         ordered = sorted(self.map_item,
-                         key=lambda mi: TypeMapItem.STRING_DATA_ITEM != mi.get_type())
-        # TODO: There could be some speedup if the parsing needs to be done only
-        # once.
-        # The idea is to parse all items in the correct order, which is possible
-        # as all items construct an acyclic graph of dependencies.
-        #
-        # We know, that we do not need to parse header_item and map_list (again)
-        # Then the following order would probably work:
-        # * string_data_item
-        # * string_id_item
-        # * type_id_item
-        # * type_list
-        # * field_id_item
-        # * proto_id_item
-        # * method_id_item
-        # * debug_info_item
-        # * code_item
-        # * method_handle_item
-        # * call_site_id_item
-        # * class_data_item
-        # * encoded_array_item
-        # * annotation_item
-        # * annotation_set_item
-        # * annotation_set_ref_item
-        # * annotations_directory_item
-        # * class_def_item
+                        key=lambda mi: load_order[TypeMapItem(mi.get_type()).value])
```

Comment:
I think `key=lambda mi: load_order[mi.get_type()]` should work as well - thats the beauty of IntEnum